content/en/platform/corda/4.12/community/comm-upgrade-guide.md

```diff
@@ -0,0 +1,275 @@
+---
+aliases:
+- /head/comm-upgrade-guide.html
+- /HEAD/comm-upgrade-guide.html
+- /comm-upgrade-guide.html
+date: '2024-06-05'
+menu:
+  corda-community-4-12:
+    identifier: corda-community-4-12-comm-upgrade-guide
+    parent: corda-community-4-12-upgrading
+    weight: 15
+tags:
+- upgrade
+- guide
+title: Corda Community Edition 4.11 to 4.12 upgrade guide
+---
+
+# Corda Community Edition 4.11 to 4.12 upgrade guide
+
+This upgrade guide outlines the steps for migrating your Corda 4.11 node to version 4.12 while maintaining backwards compatibility with earlier Corda 4.x versions.
+
+{{< note >}}
+The steps from this guide only work for direct upgrades from Corda 4.11 to 4.12. If you have any nodes on versions 4.10 and below, you must upgrade them to 4.11 first. To do that, consult the relevant release upgrade documentation.
+{{< /note >}}
+
+### Background
+
+Corda 4.12 has been upgraded to use Kotlin 1.9 and Java 17. When designing Corda 4.12, we had to take into consideration the issue of having a backchain of transactions that had been verified using Kotlin 1.2 and Java 8 by earlier versions of Corda.
+
+If a Corda 4.12 node were to be part of a network of mixed Corda 4.x versions, a Corda 4.12 node would not be able to verify a contract compiled using Kotlin 1.2. Similarly, a Corda node running an earlier version of Kotlin would not be able to verify a new Corda 4.12 contract compiled using Kotlin 1.9.
```

Comment:
"Corda 4.x" should be "Corda 4.11" since it was previously stated "The steps from this guide only work for direct upgrades from Corda 4.11 to 4.12"
There are more occurrences of "4.x" throughout.

Reply:
Here it says "a network of mixed Corda 4.x versions". I assumed Corda 4.12 node can be part of a mixed network not only with 4.11 nodes but also lower version nodes, is that correct? @chriscochrane

Reply:
Ah actually that's a good point; I suppose you can have old nodes of ANY old 4.x versions mixed in with Corda 4.12, but you can only UPGRADE to 4.12 FROM 4.11. So "4.x" is actually OK.